directory/directory_dest.go

```diff
@@ -193,6 +193,9 @@ func (d *dirImageDestination) TryReusingBlobWithOptions(ctx context.Context, inf
 	if info.Digest == "" {
 		return false, private.ReusedBlob{}, fmt.Errorf("Can not check for a blob with unknown digest")
 	}
+	if options.RequiredCompression != nil && (options.RequiredCompression != info.CompressionAlgorithm) {
```

Comment:
I think it would be easier if the new option only applied in the `CanSubstitute` case; then these simple implementations that never substitute blobs wouldn’t need to add extra conditions.

Reply:
Removed for transports which are not using `CanSubstitute`